### PR TITLE
return empty vol map if /dev/disk/by-id not there

### LINKED ISSUE #466: ScaleIO on RHEL 7.3 in AWS: /dev/disk/by-id does not exist
# Summary
I am running RHEL 7.3 in AWS and it looks like this PR https://github.com/codedellemc/libstorage/pull/414 introduced the issue. In my environment, the /dev/disk/by-id does not exist. I am not sure if that is an AWS thing or if it is an RHEL7.X thing.

```
[root@ip-10-0-0-12 libstorage]# ls -la /dev/disk
total 0
drwxr-xr-x.  4 root root   80 Mar 15 11:08 .
drwxr-xr-x. 18 root root 2760 Mar 15 11:09 ..
drwxr-xr-x.  2 root root   80 Mar 15 11:08 by-partuuid
drwxr-xr-x.  2 root root   60 Mar 15 11:08 by-uuid
``` 

## Expected Behavior
ScaleIO currently is broken in this environment.

## Actual Behavior
Should be able to retrieve volume info in a ScaleIO environment 

## Steps To Reproduce
1. Create an AWS RHEL 7.3 environment (or use the manually triggered test harness)
2. configure REX-Ray config.yml
3. Run rexray volume ls. 

## Configuration Files
```
rexray:
  logLevel: warn
  modules:
    rexray:
      type: docker
      libstorage:
        service: scaleio
      host: unix:///run/docker/plugins/rexray.sock
libstorage:
  service: scaleio
  integration:
    volume:
      operations:
        mount:
          preempt: true
scaleio:
  endpoint: https://10.0.0.11/api
  insecure: true
  thinOrThick: ThinProvisioned
  userName: admin
  password: XXXXXX
  systemId: 0a79118e5ecc882c
  protectionDomainName: default
  storagePoolName: default
```

## Logs
Running a rexray volume ls:
```
INFO[0000] created libStorage client                     host=unix:///var/run/libstorage/708268698.sock integrationDriver=linux osDriver=linux service=scaleio storageDriver=libstorage time=1489595193503
DEBU[0000] waiting on executor lock                      host=unix:///var/run/libstorage/708268698.sock instanceID=scaleio:scaleio=0a2d845900000002 integrationDriver=linux osDriver=linux server=island-mouse-jp service=scaleio storageDriver=libstorage time=1489595193503 txCR=1489595193 txID=209cb480-63f1-4fa5-5453-bfd731bc0230
DEBU[0000] invoking executor cli                         args=[scaleio localDevices quick] cmd=/var/lib/libstorage/lsx-linux host=unix:///var/run/libstorage/708268698.sock instanceID=scaleio:scaleio=0a2d845900000002 integrationDriver=linux osDriver=linux server=island-mouse-jp service=scaleio storageDriver=libstorage time=1489595193503 txCR=1489595193 txID=209cb480-63f1-4fa5-5453-bfd731bc0230
DEBU[0000] signalling executor lock                      host=unix:///var/run/libstorage/708268698.sock instanceID=scaleio:scaleio=0a2d845900000002 integrationDriver=linux osDriver=linux server=island-mouse-jp service=scaleio storageDriver=libstorage time=1489595193571 txCR=1489595193 txID=209cb480-63f1-4fa5-5453-bfd731bc0230
FATA[0000] error executing xcli                          args=[scaleio localDevices quick] inner=exit status 1 lsx=/var/lib/libstorage/lsx-linux
```

If i try to manually run /var/lib/libstorage/lsx-linux scaleio localDevices quick, i get:
```
[root@ip-10-0-0-12 libstorage]# ./lsx-linux scaleio localDevices deep
error: error getting localDevices: open /dev/disk/by-id: no such file or directory
```

Then manually inspect /dev/disk:
```
[root@ip-10-0-0-12 libstorage]# ls -la /dev/disk
total 0
drwxr-xr-x.  4 root root   80 Mar 15 11:08 .
drwxr-xr-x. 18 root root 2760 Mar 15 11:09 ..
drwxr-xr-x.  2 root root   80 Mar 15 11:08 by-partuuid
drwxr-xr-x.  2 root root   60 Mar 15 11:08 by-uuid
```

### PR DESCRIPTION
**Pull request by codenrhoden**
Fixes #466

### CODE REVIEW

File: drivers/storage/scaleio/executor/scaleio_executor.go [hunk 1]
```diff
@@ -88,6 +89,10 @@ func getLocalVolumeMap() (map[string]string, error) {
 	volMap := map[string]string{}
 	files, err := ioutil.ReadDir(diskIDPath)
 	if err != nil {
+		if os.IsNotExist(err) {
```
Comment (akutz) [inline drivers/storage/scaleio/executor/scaleio_executor.go]: Hi @codenrhoden,

Be aware of the [tricky nature](http://stackoverflow.com/questions/12518876/how-to-check-if-a-file-exists-in-go/22467409#22467409) of `os.IsNotExist`. The Gotil package has a function named [`FileExists`](https://github.com/akutz/gotil/blob/master/gotil.go#L155-L161) that already handles this.

Comment (akutz) [inline drivers/storage/scaleio/executor/scaleio_executor.go]: Hi @codenrhoden,

Be aware of the [tricky nature](http://stackoverflow.com/questions/12518876/how-to-check-if-a-file-exists-in-go/22467409#22467409) of `os.IsNotExist`. The Gotil package has a function named [`FileExists`](https://github.com/akutz/gotil/blob/master/gotil.go#L155-L161) that already handles this.

**Review by akutz — CHANGES_REQUESTED** [review]

**Comment by codenrhoden** [comment]
@akutz. Thanks for the heads up. not sure why I didn't use `gotil.FileExists()`, as I have used it before...

I guess because I was of the mindset to check what the error was, rather than checking if the dir exists _first_. Take another look when you can!

**Review by akutz — APPROVED** [review]

**Comment by akutz** [comment]
Hi @codenrhoden,

Mind rebasing this onto `master`? I can merge anyway, but since I know you're online, I prefer rebase when possible.

**Comment by codenrhoden** [comment]
@akutz I always prefer rebase as well. Should be good to go!

**Pull request merged**